src/com/boxboat/jenkins/library/credentials/ICredential.groovy

```diff
@@ -0,0 +1,5 @@
+package com.boxboat.jenkins.library.credentials
+
+interface ICredential {
+    def withCredentials(Map configMap, Closure closure)
```

Comment:
what is `configMap` used for?  must it be part of the interface?  it seems that any config that is needed by the implementation should be contained in the implementaiton

Reply:
Moved to implementations, added params class